library/Rules/Uuid.php

```diff
@@ -51,24 +56,27 @@ public function evaluate(mixed $input): Result
     {
         $template = $this->version ? self::TEMPLATE_VERSION : self::TEMPLATE_STANDARD;
         $parameters = ['version' => $this->version];
-        if (!is_string($input)) {
+
+        if (!is_string($input) && !($input instanceof UuidInterface)) {
             return Result::failed($input, $this, $parameters, $template);
         }
 
-        return new Result(preg_match($this->getPattern(), $input) > 0, $input, $this, $parameters, $template);
-    }
+        if (is_string($input)) {
+            try {
+                $input = RamseyUuid::fromString($input);
```

Comment:
This allows us to interpret non-formatted UUIDs as valid. I _think_ that's a good thing, but I would love to hear your thoughts